pyccel/codegen/printing/ccode.py

```diff
@@ -1487,17 +1494,48 @@ def _print_EmptyNode(self, expr):
         return ''
 
     #=================== OMP ==================
-    def _print_OMP_For_Loop(self, expr):
-        omp_expr   = str(expr.txt)
-        return '#pragma omp for{}\n{{'.format(omp_expr)
-
-    def _print_OMP_Parallel_Construct(self, expr):
-        omp_expr   = str(expr.txt)
-        return '#pragma omp {}\n{{'.format(omp_expr)
 
-    def _print_OMP_Single_Construct(self, expr):
-        omp_expr   = str(expr.txt)
-        return '#pragma omp {}\n{{'.format(omp_expr)
+    def _print_OmpCombinedAnnotatedComment(self, expr):
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined
+        clauses += str(expr.txt)
+        omp_expr = '#pragma omp'
+        if isinstance(expr, OMP_Parallel_Construct):
+            omp_expr += ' parallel{}'.format(clauses)
```

Comment:
Do you think `parallel`, `masked`, `target` and `teams` could be a property of the construct? E.g:
```
class OmpCombinedAnnotatedComment(OmpAnnotatedComment):
    """ Represents a Combined construct. """
    _is_multiline = False
    @classmethod
    def is_multiline(cls):
        return cls._is_multiline

    @property
    def keyword(self):
        return ''

class OMP_Teams_Construct(OmpCombinedAnnotatedComment):
    """ Represents OpenMP Teams construct. """
    _is_multiline = True
    @property
    def keyword(self):
        return 'teams'
```

Then you could do:
```
 def _print_OmpCombinedAnnotatedComment(self, expr):
        clauses = ''
        if expr.combined:
            clauses = ' ' + expr.combined
        clauses += str(expr.txt)
        omp_expr = '#pragma omp'
        if expr.is_multiline:
            omp_expr += ' {}{}'.format(expr.keyword, clauses)
            if expr.combined is None:
                omp_expr += '\n{'
            elif (expr.combined and "for" not in expr.combined):
                if "masked taskloop" not in expr.combined:
                    omp_expr += '\n{'
        return omp_expr
```